app/Console/Commands/Members/ImportMembers.php

```diff
@@ -4,214 +4,102 @@
-
-            //     return;
-            // }
-
-            // if (isset($_prevRat->PreviousRatingInt)) {
-            //     $prevAtcRating = Qualification::parseVatsimATCQualification($_prevRat->PreviousRatingInt);
-
-            //     if ($prevAtcRating) {
-            //         $member->addQualification($prevAtcRating);
-            //     }
-            // }
-        }
+        $this->info('Fetching members from VATSIM API...');
 
-        // if they're a division member, or their current rating isn't instructor, log their 'main' rating
-        if (($member_data['rating_atc'] < 8) || $member->hasState('DIVISION')) {
-            $atcRating = Qualification::parseVatsimATCQualification($member_data['rating_atc']);
+        $response = Http::withHeaders([
+            'Authorization' => "Token {$this->apiToken}",
+        ])->get(config('vatsim-api.base').'divisions/GBR/members');
 
-            if ($atcRating) {
-                $member->addQualification($atcRating);
-            }
+        // Process first page of results
+        foreach ($response->collect()->get('results') as $result) {
+            $this->importedMembers->push($result);
         }
 
-        // anything else is processed by the Members:CertUpdate script
+        // Process paginated results
+        while ($response->successful() && $response->collect()->get('next') != null) {
+            $response = Http::withHeaders([
+                'Authorization' => "Token {$this->apiToken}",
+            ])->get($response->collect()->get('next'));
 
-        if ($member->hasState('DIVISION') && $member->email) {
-            $member->notify(new WelcomeMember());
+            foreach ($response->collect()->get('results') as $result) {
+                $this->importedMembers->push($result);
+            }
         }
-    }
-
-    protected function updateMember($member_data)
-    {
-        $member = Account::find($member_data['cid']);
-        $member->name_first = $member_data['name_first'];
-        $member->name_last = $member_data['name_last'];
-        $member->email = $member_data['email'];
-        $member->save();
 
-        $member->addState(State::findByCode('DIVISION'), 'EMEA', 'GBR');
+        $this->info("{$this->importedMembers->count()} members obtained from VATSIM API.");
     }
 
-    protected function getMemberIdAndEmail()
+    protected function processMember(array $member)
     {
-        return DB::table('mship_account')
-            ->pluck('email', 'id');
+        $account = Account::updateOrCreate(
+            ['id' => $member['id']],
+            [
+                'name_first' => $member['name_first'],
+                'name_last' => $member['name_last'],
+                'email' => $member['email'],
+                'joined_at' => Carbon::create($member['reg_date']),
+                'inactive' => (int) $member['rating'] < 0,
+            ]
+        );
+
+        $account->wasRecentlyCreated ?? $account->notify(new WelcomeMember());
```

Comment:
Worth queueing this rather than dispatching the notification directly from the command?

Reply:
Its queued on the Notification class